feature/experimental/gribi/otg_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,703 @@
+		WithAFT(fluent.AllAFTs).
+		Send()
+
+	if err != nil {
+		t.Fatalf("Unexpected error from get, got: %v", err)
+	}
+	return len(gr.GetEntry())
+}
+
+// Send traffic and validate traffic.
+func testTraffic(t *testing.T, args testArgs) {
+	t.Helper()
+	args.ate.OTG().StartTraffic(t)
+	// Send traffic for 1 minute.
+	time.Sleep(1 * time.Minute)
+	args.ate.OTG().StopTraffic(t)
+	otgutils.LogFlowMetrics(t, args.ate.OTG(), args.top)
+
+	txPkts := float32(gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Flow("Flow").Counters().OutPkts().State()))
+	rxPkts := float32(gnmi.Get(t, args.ate.OTG(), gnmi.OTG().Flow("Flow").Counters().InPkts().State()))
+	if txPkts == 0 {
+		t.Fatalf("Tx packets should be higher than 0")
+	}
+
+	if got := (txPkts - rxPkts) * 100 / txPkts; got > 0 {
+		t.Errorf("LossPct got %f, want 0", got)
+	} else {
+		t.Logf("Traffic flows fine from ATE-port1 to ATE-port2.")
+	}
+}
+
+// validateSwitchoverStatus is to validate switchover status.
+func validateSwitchoverStatus(t *testing.T, dut *ondatra.DUTDevice, secondaryBeforeSwitch string) string {
+	t.Helper()
+	startSwitchover := time.Now()
+	t.Logf("Wait for new Primary controller to boot up by polling the telemetry output.")
+	for {
+		var currentTime string
+		t.Logf("Time elapsed %.2f seconds since switchover started.", time.Since(startSwitchover).Seconds())
+		time.Sleep(30 * time.Second)
+		if errMsg := testt.CaptureFatal(t, func(t testing.TB) {
+			currentTime = gnmi.Get(t, dut, gnmi.OC().System().CurrentDatetime().State())
+		}); errMsg != nil {
+			t.Logf("Got testt.CaptureFatal errMsg: %s, keep polling ...", *errMsg)
+		} else {
+			t.Logf("Controller switchover has completed successfully with received time: %v", currentTime)
+			break
+		}
+		if uint64(time.Since(startSwitchover).Seconds()) > maxSwitchoverTime {
+			t.Fatalf("time.Since(startSwitchover): got %v, want < %v", time.Since(startSwitchover), maxSwitchoverTime)
+		}
+	}
+	t.Logf("Controller switchover time: %.2f seconds", time.Since(startSwitchover).Seconds())
+	// Old secondary controller becomes primary after switchover.
+	return secondaryBeforeSwitch
+}
+
+// incrementMAC increments the MAC by i. Returns error if the mac cannot be parsed or overflows the mac address space
+func incrementMAC(mac string, i int) (string, error) {
+	macAddr, err := net.ParseMAC(mac)
+	if err != nil {
+		return "", err
+	}
+	convMac := binary.BigEndian.Uint64(append([]byte{0, 0}, macAddr...))
+	convMac = convMac + uint64(i)
+	buf := new(bytes.Buffer)
+	err = binary.Write(buf, binary.BigEndian, convMac)
+	if err != nil {
+		return "", err
+	}
+	newMac := net.HardwareAddr(buf.Bytes()[2:8])
+	return newMac.String(), nil
+}
+
+// Waits for an ARP entry to be present for ATE Port1
+func waitOTGARPEntry(t *testing.T) {
+	t.Helper()
+	ate := ondatra.ATE(t, "ate")
+	gnmi.WatchAll(t, ate.OTG(), gnmi.OTG().Interface(atePort1.Name+".Eth").Ipv4NeighborAny().LinkLayerAddress().State(), time.Minute, func(val *ygnmi.Value[string]) bool {
```

Comment:
This isn't the interface name that's configured (this value is "atePort1.Eth", vs the configured "src.Eth"). "src" does provide clarity, can we make that a constant in the main test function and provide it as an argument here?

Reply:
The ARP has to be verified on the transmitting interface which will always be atePort1 and its ethernet device "atePort1.Eth". We don't need to do this on any other interface.

Reply:
I'm reading line 497 (https://github.com/openconfig/featureprofiles/pull/1266/files#diff-8fe31392881dd82fdf05ddff5b1fa8cff8d77deab519c9a3e701934d139c5523R497) as configuring the interface name is "src.Eth" since the 'name' arg provided to configureATE is 'src'. (I agree on only checking the transmitting interface, I think it may not be named correctly though.

Reply:
You're right..there was an incorrect input to configureATE there so I've changed it from "src" to atePort1.Name